### DISCUSSION TITLE
Server Just Crashes (tShock)

### DISCUSSION THREAD

**Discussion by YKKULTRA** [Pryaxis/TShock#2502] [Q&A] [answered]
11/21/2021 5:53:47 PM
System.NullReferenceException: Object reference not set to an instance of an object.
   at Terraria.Main.WriteFancyWorldLoadErrorToConsole()
   at Terraria.Main.DedServ()
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs)

The world doesn't work on tShock.
But it works fine in vanilla server.
When i create a world form tShock it can launch and seem to appear in worlds folder.

**Comment by Arthri** [comment, marked as answer]
Your `OTAPI.dll` is not properly updated/is wrong

**Reply by YKKULTRA** [reply]
i downloaded it again and it didnt work (tried replacing with new one)

**Reply by Arthri** [reply]
Where are you downloading it from? TShock hasn't officially updated to 1.4.3 yet so you will have to use a pre-pre-release build: https://github.com/Pryaxis/TShock/actions/runs/1483941680

Direct download link: https://github.com/Pryaxis/TShock/suites/4415266702/artifacts/116716447

**Reply by YKKULTRA** [reply]
i didnt dowload the prerelease i just downloaded the latest

**Reply by YKKULTRA** [reply]
it did work thx
